<?xml version="1.0"?>
<launch>
  <arg name="input_objects" default="/perception/object_recognition/objects"/>
  <arg name="input_pointcloud" default="/perception/obstacle_segmentation/pointcloud"/>
  <arg name="input_approval" default="/external/approval"/>
  <arg name="output_filtered_objects" default="/perception/object_recognition/filtered_objects"/>
  <arg name="output_filtered_pointcloud" default="/perception/obstacle_segmentation/filtered_pointcloud"/>
  <arg name="perception_filter_param_path" default="$(find-pkg-share autoware_perception_filter)/config/perception_filter.param.yaml"/>

  <node pkg="autoware_perception_filter" exec="autoware_perception_filter_node" name="perception_filter_node" output="screen">
    <remap from="input/objects" to="$(var input_objects)"/>
    <remap from="input/pointcloud" to="$(var input_pointcloud)"/>
    <remap from="input/approval" to="$(var input_approval)"/>
    <remap from="output/filtered_objects" to="$(var output_filtered_objects)"/>
    <remap from="output/filtered_pointcloud" to="$(var output_filtered_pointcloud)"/>
    <param from="$(var perception_filter_param_path)"/>
  </node>
</launch>
